examples/imdb_fasttext.py

```diff
@@ -21,7 +22,49 @@
 from keras.datasets import imdb
 
 
+def create_ngram_set(input_list, ngram_range=2):
+    """
+    Extract n-grams from a list of integers
+
+    >>> create_ngram_set([1, 4, 9, 4, 1], ngram_range=2)
+    {(4, 9), (4, 1), (1, 4), (9, 4)}
+
+    :param input_list:
+    :param ngram_range: integer
+    :return: set of tuples
+    """
+    return set(zip(*[input_list[i:] for i in range(ngram_range)]))
+
+
+def add_ngram(sequences, token_indice, ngram_range=2):
+    """
+    Augment the list of sequences by appending bi-grams terms
+
+    >>> print(add_ngram([[1, 3, 4, 5], [1, 3, 7, 9, 2]],
+        token_indice={(1, 3): 1337, (9, 2): 42, (4, 5): 2017},
+        ngram_range=2))
+    [[1, 3, 4, 5, 1337, 2017], [1, 3, 7, 9, 2, 1337, 42]]
+
+    :param sequences:
+    :param token_indice:
+    :param ngram_range:
+    :return:
+    """
+    augmented_sequences = []
+
+    for sequence in sequences.copy():
+        for i in range(len(sequence)-ngram_range+1):
+            ngram = tuple(sequence[i:i+ngram_range])
+            if ngram in token_indice:
+                sequence.append(token_indice[ngram])
+        augmented_sequences.append(sequence)
+
+    return augmented_sequences
+
+
 # set parameters:
+UNIGRAM = True
+BIGRAM = False
```

Comment:
This can easily be generalized to n-grams with one single preprocessing function. A more elegant approach would be preferable.